lib/smart_proxy_ansible/playbooks_reader.rb

```diff
@@ -0,0 +1,31 @@
+module Proxy
+  module Ansible
+    class PlaybooksReader
+      class << self
+        def playbooks_names
+          ReaderHelper.collections_paths.split(':').map { |path| read_collection_playbooks(path).map { |playbook| playbook[:name] } }.flatten
+        end
+
+        def playbooks(playbooks_to_import)
+          ReaderHelper.collections_paths.split(':').reduce([]) do |playbooks, path|
+            playbooks.push(read_collection_playbooks(path, playbooks_to_import: playbooks_to_import))
+            end.flatten
+        end
+
+        def read_collection_playbooks(collections_path, playbooks_to_import: nil)
+          Dir.glob("#{collections_path}/ansible_collections/*/*/playbooks/*").map do |path|
+              name = ReaderHelper.playbook_or_role_full_name(path)
+              {
+                  name: name,
+                  playbooks_content: File.readlines(path)
+              } if playbooks_to_import.nil? || playbooks_to_import.include?(name)
+          end.compact
+        rescue Errno::ENOENT, Errno::EACCES => e
+          logger.debug(e.backtrace)
+          message = "Could not read Ansible playbooks #{collections_path} - #{e.message}"
+          raise ReadPlaybooksException.new(message), message
```

Comment:
this seems to set the message redundantly, I do not think the second argument is needed IMHO